### PR TITLE
Develope page manage farmer

### PR DESCRIPTION
**Pull request by HaManhDong**
- Add action crud and check role
- Todo: add class for button submit in form and fix select2 selected data in edit form

### CODE REVIEW

File: pgscm/farmer/views.py [hunk 1]
```diff
@@ -22,35 +26,84 @@ def index():
...
 
         return render_template('farmer/index.html', farmers=farmers,
                                form=form, dform=dform)
+
+
+def check_role():
+    for r in crud_role:
+        if r == current_user.roles[0].name:
+            return True
+    flash('You have no permission!', 'warning')
```
Comment (hieulq) [inline pgscm/farmer/views.py]: i18n, the correct is `flash('warning', 'no permission')` and delete block of code in template regarding flash message.

File: pgscm/farmer/views.py [hunk 2]
```diff
@@ -22,35 +26,84 @@ def index():
...
+                farmers.remove(del_farmer)
+                flash('Delete farmer success!', 'info')
+
+                return redirect(url_for(request.endpoint))
 
         return render_template('farmer/index.html', farmers=farmers,
                                form=form, dform=dform)
+
+
+def check_role():
```
Comment (hieulq) [inline pgscm/farmer/views.py]: this method should move to utils for reusing.

File: pgscm/templates/farmer/index.html [hunk 3]
```diff
@@ -27,8 +27,21 @@
...
+
+{% with messages = get_flashed_messages(with_categories=true) %}
+    {% if messages %}
+        <ul class=flashes>
+            {% for category, message in messages %}
+                <li class="{{ category }}">{{ message }}</li>
+            {% endfor %}
+        </ul>
+    {% endif %}
+{% endwith %}
```
Comment (hieulq) [inline pgscm/templates/farmer/index.html]: this block of code is not necessary because we have toastr enabled

File: pgscm/utils.py [hunk 4]
```diff
@@ -24,8 +24,17 @@ def __call__(self, field, **kwargs):
         return super(Select, self).__call__(field, **kwargs)
 
 
+# class CustomSubmitField(SubmitField):
+#     def __call__(self, **kwargs):
+#         c = kwargs.pop('class', '') or kwargs.pop('class_', '')
+#         kwargs['class'] = c + " " + const.SUBMIT_DEFAULT_CLASS
+#         return super(CustomSubmitField, self).__call__(**kwargs)
```
Comment (hieulq) [inline pgscm/utils.py]: delete it

**Review by hieulq — CHANGES_REQUESTED** [review]

File: pgscm/utils.py [hunk 5]
```diff
@@ -24,17 +26,18 @@ def __call__(self, field, **kwargs):
...
         render_kw={
             "placeholder": __('Describe your reasons to delete this data')})
     submit_del = SubmitField(__('Delete'))
+
+
+def check_role(roles):
+    for r in roles:
+        if r == current_user.roles[0].name:
+            return True
+    flash('You have no permission!', 'warning')
```
Comment (hieulq) [inline pgscm/utils.py]: i18n for message?

**Review by hieulq — CHANGES_REQUESTED** [review]

File: pgscm/utils.py (hunk 5 above)
Comment (HaManhDong) [reply pgscm/utils.py]: Yes, sory i miss it.

**Review by hieulq — APPROVED** [review]

**Pull request merged**